src/JWT/Serializers/JsonNetSerializer.cs

```diff
@@ -8,22 +9,34 @@ namespace JWT.Serializers
     /// </summary>
     public sealed class JsonNetSerializer : IJsonSerializer
     {
+        private readonly JsonSerializer _serializer;
+
         /// <summary>
-        /// Serialize the given object.
+        /// Creates a new instance of <see cref="JsonNetSerializer" />.
         /// </summary>
-        /// <param name="obj">The object to serialize.</param>
-        /// <returns></returns>
-        public string Serialize(object obj)
+        /// <remarks>Uses <see cref="JsonSerializer.CreateDefault()" /> as internal serializer.</remarks>
+        public JsonNetSerializer()
+            : this(JsonSerializer.CreateDefault())
         {
-            return JObject.FromObject(obj).ToString(Formatting.None);
+
         }
 
         /// <summary>
-        /// Deserialize the given string.
+        /// Creates a new instance of <see cref="JsonNetSerializer" />.
         /// </summary>
-        /// <typeparam name="T">The type to deserialize the string to.</typeparam>
-        /// <param name="json">The JSON to be deserialized.</param>
-        /// <returns></returns>
+        /// <param name="serializer">Internal <see cref="JsonSerializer" /> to use for serialization.</param>
+        public JsonNetSerializer(JsonSerializer serializer)
+        {
+            _serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
+        }
+
+        /// <inheritdoc />
+        public string Serialize(object obj)
+        {
+            return JObject.FromObject(obj, _serializer).ToString(Formatting.None);

```

Comment:
@abatishchev I feel like the formatting is ignored during the `FromObject` stage. And unfortunately, needs to be applied to the `ToString()` stage.

so ...

```
return JObject.FromObject(obj, _serializer).ToString(_serializer.Formatting);
```

BUT! because of me checking out that overload in the `ToString()` .. there's also a converters input param... which means this is also valid.

```
return JObject.FromObject(obj, _serializer).ToString(_serializer.Formatting, _serializer.Converters.ToArray());
```

This is me testing various permutations over the last 30-60 mins. 
I've also ref'd these: 
- https://stackoverflow.com/questions/570689/how-to-serialize-a-jobject-without-the-formatting
- https://stackoverflow.com/questions/20857432/json-net-jobject-fromobject-vs-jsonconvert-deserializeobjectjobjectjsonconver